### PR TITLE
Fix compilation warnings

### LINKED ISSUE #35: Compilation warnings
Hello,

I compiled your code on `Ubuntu 16.04.3 LTS` and I saw many warnings (I am able to fix it, so stay tuned):

```bash
celestian@vega:~/demoinfogo-linux$ make
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demofile.cpp -o src/demofile.o
src/demofile.cpp: In member function ‘bool CDemoFile::Open(const char*)’:
src/demofile.cpp:150:56: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]
   fread( &m_DemoHeader, 1, sizeof( m_DemoHeader ), fp );
                                                        ^
src/demofile.cpp:168:45: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]
   fread( &m_fileBuffer[ 0 ], 1, Length, fp );
                                             ^
mkdir -p src/generated_proto
protobuf-2.5.0/src/protoc --proto_path=./src --proto_path=protobuf-2.5.0/src --cpp_out=./src/generated_proto src/steammessages.proto
mkdir -p src/generated_proto
protobuf-2.5.0/src/protoc --proto_path=./src --proto_path=protobuf-2.5.0/src --cpp_out=./src/generated_proto src/cstrike15_gcmessages.proto
mkdir -p src/generated_proto
protobuf-2.5.0/src/protoc --proto_path=./src --proto_path=protobuf-2.5.0/src --cpp_out=./src/generated_proto src/netmessages_public.proto
mkdir -p src/generated_proto
protobuf-2.5.0/src/protoc --proto_path=./src --proto_path=protobuf-2.5.0/src --cpp_out=./src/generated_proto src/cstrike15_usermessages_public.proto
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demofiledump.cpp -o src/demofiledump.o
In file included from src/demofiledump.h:30:0,
                 from src/demofiledump.cpp:27:
src/demofilepropdecode.h: In member function ‘void Prop_t::Print(int)’:
src/demofilepropdecode.h:88:10: warning: enumeration value ‘DPT_NUMSendPropTypes’ not handled in switch [-Wswitch]
   switch ( m_type )
          ^
src/demofiledump.cpp: In function ‘void ParseStringTableUpdate(CBitRead&, int, int, int, int, int, bool)’:
src/demofiledump.cpp:678:17: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if ( nBytes > sizeof( tempbuf ) )
                 ^
src/demofiledump.cpp:728:73: warning: format ‘%s’ expects argument of type ‘char*’, but argument 5 has type ‘const void*’ [-Wformat=]
     printf( " %d, %s, %d, %s \n", entryIndex, pEntry, nBytes, pUserData );
                                                                         ^
src/demofiledump.cpp:595:6: warning: unused variable ‘lastDictionaryIndex’ [-Wunused-variable]
  int lastDictionaryIndex = -1;
      ^
src/demofiledump.cpp: In function ‘CSVCMsg_SendTable* GetTableByClassID(uint32)’:
src/demofiledump.cpp:835:38: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if ( s_ServerClasses[ i ].nClassID == nClassID )
                                      ^
src/demofiledump.cpp: In function ‘void FlattenDataTable(int)’:
src/demofiledump.cpp:1003:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (prop->priority() == priority || (priority == 64 && (SPROP_CHANGES_OFTEN & prop->flags()))) 
                          ^
src/demofiledump.cpp: In function ‘void PrintNetMessage(CDemoFileDump&, const void*, int) [with T = CSVCMsg_PacketEntities; int msgType = 26]’:
src/demofiledump.cpp:1159:7: warning: unused variable ‘nBaseline’ [-Wunused-variable]
   int nBaseline = msg.baseline();
       ^
src/demofiledump.cpp:1160:8: warning: unused variable ‘bUpdateBaselines’ [-Wunused-variable]
   bool bUpdateBaselines = msg.update_baseline();
        ^
src/demofiledump.cpp: In function ‘bool ParseDataTable(CBitRead&)’:
src/demofiledump.cpp:1453:7: warning: unused variable ‘type’ [-Wunused-variable]
   int type = buf.ReadVarInt32();
       ^
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demoinfogo.cpp -o src/demoinfogo.o
In file included from src/demofiledump.h:30:0,
                 from src/demoinfogo.cpp:25:
src/demofilepropdecode.h: In member function ‘void Prop_t::Print(int)’:
src/demofilepropdecode.h:88:10: warning: enumeration value ‘DPT_NUMSendPropTypes’ not handled in switch [-Wswitch]
   switch ( m_type )
          ^
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demofilebitbuf.cpp -o src/demofilebitbuf.o
src/demofilebitbuf.cpp: In member function ‘float CBitRead::ReadBitFloat()’:
src/demofilebitbuf.cpp:691:32: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  return *( ( float * ) &nvalue );
                                ^
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demofilepropdecode.cpp -o src/demofilepropdecode.o
In file included from src/demofiledump.h:30:0,
                 from src/demofilepropdecode.cpp:25:
src/demofilepropdecode.h: In member function ‘void Prop_t::Print(int)’:
src/demofilepropdecode.h:88:10: warning: enumeration value ‘DPT_NUMSendPropTypes’ not handled in switch [-Wswitch]
   switch ( m_type )
          ^
g++ -Iprotobuf-2.5.0/src src/demofile.o src/demofiledump.o src/demoinfogo.o src/demofilebitbuf.o src/demofilepropdecode.o src/generated_proto/steammessages.pb.cc src/generated_proto/cstrike15_gcmessages.pb.cc src/generated_proto/netmessages_public.pb.cc src/generated_proto/cstrike15_usermessages_public.pb.cc protobuf-2.5.0/src/.libs/libprotobuf.a -m32 -pthread -static -o demoinfogo
```

### PR DESCRIPTION
**Pull request by celestian**
Fixes #35

### CODE REVIEW

File: src/demofile.cpp [hunk 1]
```diff
@@ -147,7 +150,7 @@ bool CDemoFile::Open( const char *name )
 			return false;
 		}
 
-		fread( &m_DemoHeader, 1, sizeof( m_DemoHeader ), fp );
+		readed = fread( &m_DemoHeader, 1, sizeof( m_DemoHeader ), fp );
```
Comment (kaimallea) [inline src/demofile.cpp]: `readed` sounds weird. is there a better variable name for this?

**Comment by kaimallea** [comment]
@celestian you are amazing, sir. thanks for your PR. one nitpick comment

File: src/demofile.cpp (hunk 1 above)
Comment (celestian) [reply src/demofile.cpp]: @kaimallea `fread()` returns size of read bytes. So... `read_size` would be ok?

**Comment by celestian** [comment]
I addressed you nitpick. If you have better name on top of your mind, please tell me.

I would like to write some statistics on `.dem` file. This project is good starting point. I know, it is trying to be the same as original project for Windows. So... I will start new one. But I still need this because `.dem` format is not so obvious :-)

**Comment by kaimallea** [comment]
@celestian looks good, thank you very much for your contribution! 👍

**Review by kaimallea — APPROVED** [review]
🚀

**Pull request merged**